app/models/room.rb

```diff
@@ -68,6 +68,15 @@ def self.search_with(params)
     Room.where(where_query)
   end
 
+  def self.sort_with(params)
+    sort_by   = params.fetch("sort_by", "name||").split("||")
+    sort_dir  = params.fetch("sort_dir", "ASC||").split("||")
+    sort_hash = Hash[sort_by.zip(sort_dir)]
+
+    order_query = sort_hash.map { |sort_by, sort_dir| "#{sort_by} #{sort_dir}" }.join(", ")
+    self.order(order_query)
```

Comment:
Why all the fetching and splitting and zipping and mapping and joining?

We only have two possible sorting directions `asc` and `desc`, so we can default to one of those unless specified otherwise.

We can only sort on columns that are columns in the database, so the sort_by param should really be validated at some point.

```
def self.sort_with(params)
  if self.column_names.include?(params[:sort_by])
    sort_dir = params[:sort_dir] == 'asc' ? 'asc' : 'desc'
    self.order( "#{params[:sort_by]} #{sort_dir}")
  else
    self.order
  end
end
```

This could also be done a variety of others ways as well. http://railscasts.com/episodes/228-sortable-table-columns